jetty-websocket/websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/util/AutoDemandingMessageHandler.java

```diff
@@ -0,0 +1,186 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.websocket.core.util;
+
+import java.nio.ByteBuffer;
+
+import org.eclipse.jetty.io.ByteBufferAccumulator;
+import org.eclipse.jetty.util.BufferUtil;
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.Utf8Appendable;
+import org.eclipse.jetty.util.Utf8StringBuilder;
+import org.eclipse.jetty.websocket.core.CloseStatus;
+import org.eclipse.jetty.websocket.core.Frame;
+import org.eclipse.jetty.websocket.core.FrameHandler;
+import org.eclipse.jetty.websocket.core.exception.BadPayloadException;
+import org.eclipse.jetty.websocket.core.exception.MessageTooLargeException;
+
+/**
+ * <p>
+ * A utility implementation of FrameHandler that de-fragments text frames and binary frames into a whole messages before
+ * calling {@link #onText(String, Callback)} or {@link #onBinary(ByteBuffer, Callback)}.
+ * </p>
+ * <p>
+ * This is a demanding frame handler, meaning {@link FrameHandler#isDemanding()} returns false.
```

Comment:
OMG! This sentence is nuts, just remove it.

Can you explain what this class does without going into these much confusing details?
Something like "This class implicitly demands more frames when the `Callback` passed to either `onText()` or `onBinary()` is succeeded."

Please open an issue to rename this insanity in Jetty 12 😬